desc/grid.py

```diff
@@ -1885,8 +1887,13 @@ def _periodic_spacing(x, period=2 * jnp.pi, sort=False, jnp=jnp):
         x = jnp.sort(x, axis=0)
     # choose dx to be half the distance between its neighbors
     if x.size > 1:
-        dx_0 = x[1] + (period - x[-1]) % period
-        dx_1 = x[0] + (period - x[-2]) % period
+        if np.isfinite(period):
+            dx_0 = x[1] + (period - x[-1]) % period
+            dx_1 = x[0] + (period - x[-2]) % period
+        else:
+            # just set to 0 to stop nan gradient, even though above gives expected value
```

Comment:
So `np.inf % np.inf` yields `np.nan`, and any operation with an nan propogates the nan. Hence, whenever an `np.nan` occurs in a computation, they tend to consume all the gradient information.

Even though the logically correct value for dx_0 and dx_1 is np.nan or np.inf here, setting it to that value is not a good idea since non-finite values tend to consume consume the gradient information of AD tools.

This doesn't change anything for the bounce integral differentiation, since we never use `source_grid.spacing[:, 2]` in the computation where this `nan` used to be. However, JAX's nan debugger will flag any nan generated, even if it doesn't leak into the gradient, so this change also makes debugging easier.